### PR TITLE
Yet more cmake updates

### PR DESCRIPTION
**Pull request by phlptp**
@nightlark I think this updates the HELICSConfig.cmake appropriately.  It also installs some of the libzmq stuff on Windows when it is supposed to.  
I also added an option for AUTOBUILD_INSTALL_PATH to change where the autobuild stuff gets built and found.

### CODE REVIEW

**Comment by phlptp** [comment]
PR #203 needs to be merged before this one

File: CMakeLists.txt [hunk 1]
```diff
@@ -361,6 +350,14 @@ endif(BUILD_HELICS_EXAMPLES)
 INSTALL(FILES ${KEY_LIBRARY_FILES}
 	DESTINATION ${CMAKE_INSTALL_BINDIR}
 	COMPONENT runtime)
+
+install(DIRECTORY ${AUTOBUILD_INSTALL_PATH}/lib/ DESTINATION ${CMAKE_INSTALL_LIBDIR}/ OPTIONAL)
+install(DIRECTORY ${AUTOBUILD_INSTALL_PATH}/lib64/ DESTINATION ${CMAKE_INSTALL_LIBDIR}/ OPTIONAL)
+
+if (HELICS_HAVE_ZEROMQ)
+	install(FILES $<TARGET_LINKER_FILE:libzmq> DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT libraries)
```
Comment (nightlark) [inline CMakeLists.txt]: This causes problems when a system copy of ZeroMQ is used. On LC this copied the libzmq.so symlink, which points to libzmq.so.5.0.1; I think for system copies of ZeroMQ, we probably don't want to copy it, or if we do we need to copy the file pointed to by the symlink (which could be another symlink).

**Comment by nightlark** [comment]
For target names, is there one that could be exported as "Helics::Helics" as the main one for users to link with instead of Helics::helics-static or Helics::helics_base? It could be set up as an alias to one of the other targets. Another thought was using HELICS:: as the namespace instead of Helics::, but both forms of the name seem to be used elsewhere.

**Comment by phlptp** [comment]
we can add an alias easy enough.  I guess helics-static would make the most sense.  do you think all capital letters makes sense for the namespace?

File: CMakeLists.txt (hunk 1 above)
Comment (phlptp) [reply CMakeLists.txt]: I think I meant that to be restricted to windows and not msys.

File: config/HELICSConfig.cmake.in [hunk 2]
```diff
@@ -54,6 +54,8 @@ set(CMAKE_MODULE_PATH ${PACKAGE_PREFIX_DIR}/@HELICS_CMAKECONFIG_INSTALL_DIR@)
 
 include(CMakeFindDependencyMacro)
 
+include(extraMacros)
```
Comment (phlptp) [inline config/HELICSConfig.cmake.in]: Why do we need the extraMacros?

**Comment by phlptp** [comment]
Apparently ALIAS cannot be added to imported targets until CMAKE 3.11  and then only with restrictions.  I suppose we can think a bit more about what we want the targets to be called and if there are names for the existing targets that might make more sense.

File: config/HELICSConfig.cmake.in (hunk 2 above)
Comment (nightlark) [reply config/HELICSConfig.cmake.in]: The SHOW_VARIABLE macro in addBoost.cmake was defined in extraMacros — got an error without it included.

**Comment by nightlark** [comment]
Huh, that’s too bad ALIAS can’t be used for imported targets. Another thing I came across was creating aliases with the namespace prefix for exported projects, to enable projects using Helics as a subproject to link with the same target names as find_package would create.

**Comment by nightlark** [comment]
I want to try the latest changes once Quartz is up again.

**Pull request merged**